core/src/mender-client.c

```diff
@@ -805,6 +801,48 @@ mender_client_initialization_work_function(void) {
     return ret;
 }
 
+static mender_err_t
+mender_commit_artifact_data(void) {
+
+    cJSON *json_artifact_name = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "artifact_name");
+    if (NULL == json_artifact_name) {
+        mender_log_error("Unable to get artifact name from the deployment data");
+        return MENDER_FAIL;
+    }
+
+    if (MENDER_OK != mender_storage_set_artifact_name(json_artifact_name->valuestring)) {
+        mender_log_error("Unable to set artifact name");
+        return MENDER_FAIL;
+    }
+
+#ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
+#ifdef CONFIG_MENDER_PROVIDES_DEPENDS
+    /* Get provides from the deployment data */
+    cJSON *provides = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "provides");
```

Comment:
Will the provides from the deployment data contain already `artifact_name`? (I think yes, is it handled in `mender_prepare_new_provides`?) Otherwise we should add it on the fly before storing.